src/cli-constant.js

```diff
@@ -1,64 +1,227 @@
+  },
+  "no-bracket-spacing": {
+    type: "boolean",
+    isDocsOnly: true,
+    description: "Do not print spaces between brackets."
+  },
+  "no-color": {
+    type: "boolean",
+    isDocsOnly: true,
+    description: "Do not colorize error messages."
+  },
+  "no-config": {
+    type: "boolean",
+    isDocsOnly: true,
+    description: "Do not look for a configuration file."
+  },
+  "no-semi": {
+    type: "boolean",
+    isDocsOnly: true,
+    description:
+      "Do not print semicolons, except at the beginning of lines which may need them."
+  },
+  parser: {
+    type: "choice",
+    isFormatOption: true,
+    choices: ["flow", "babylon", "typescript", "postcss", "json", "graphql"],
+    description: "Specify which parse to use. Defaults to babylon."
+  },
+  "print-width": {
+    type: "int",
+    isFormatOption: true,
+    description:
+      "Specify the length of line that the printer will wrap on. Defaults to 80."
+  },
+  "range-end": {
+    type: "int",
+    isFormatOption: true,
+    description: dedent(`
+      Format code ending at a given character offset (exclusive).
+      The range will extend forwards to the end of the selected statement.
+      This option cannot be used with --cursor-offset.
+      Defaults to Infinity.
+    `)
+  },
+  "range-start": {
+    type: "int",
+    isFormatOption: true,
+    description: dedent(`
+      Format code starting at a given character offset.
+      The range will extend backwards to the start of the first line containing the selected statement.
+      This option cannot be used with --cursor-offset.
+      Defaults to 0.
+    `)
+  },
+  semi: {
+    type: "boolean",
+    isFormatOption: true,
+    isHidden: true
+  },
+  "single-quote": {
+    type: "boolean",
+    isFormatOption: true,
+    description: "Use single quotes instead of double quotes."
   },
+  stdin: {
+    type: "boolean",
+    description: "Read input from stdin."
+  },
+  "stdin-filepath": {
+    type: "path",
+    description: "Path to the file used to read from stdin."
+  },
+  "tab-width": {
+    type: "int",
+    isFormatOption: true,
+    description:
+      "Specify the number of spaces per indentation-level. Defaults to 2."
+  },
+  "trailing-comma": {
+    type: "choice",
+    isFormatOption: true,
+    choices: ["none", "es5", "all"],
+    description:
+      "Print trailing commas wherever possible when multi-line. Defaults to none."
+  },
+  "use-tabs": {
+    type: "boolean",
+    isFormatOption: true,
+    description: "Indent lines with tabs instead of spaces."
+  },
+  version: {
+    type: "boolean",
+    alias: "v",
+    description: "Print Prettier version."
+  },
+  "with-node-modules": {
+    type: "boolean",
+    description: "Process files inside 'node_modules' directory."
+  },
+  write: {
+    type: "boolean",
+    description: "Edit the file in-place. (Beware!)"
+  }
+});
+
+const optionArray = Object.keys(options).reduce(
+  (current, name) => current.concat(options[name]),
+  []
+);
```

Comment:
Isn't this better written as `Object.keys(options).map(name => options[name])`?

Reply:
Oops, forgot it's an array too. 😅